src/main/java/net/coderbot/iris/gui/property/OptionProperty.java

```diff
@@ -0,0 +1,111 @@
+package net.coderbot.iris.gui.property;
+
+import com.google.common.collect.Lists;
+import net.coderbot.iris.Iris;
+import net.coderbot.iris.gui.GuiUtil;
+import net.coderbot.iris.gui.element.PropertyDocumentWidget;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.text.Text;
+
+import java.util.List;
+
+public abstract class OptionProperty<T> extends ValueProperty<T> {
+    protected T[] values;
+    protected int index;
+    protected final int defaultIndex;
+    protected final boolean isSlider;
+
+    public OptionProperty(T[] values, int defaultIndex, PropertyDocumentWidget document, String key, Text label, boolean isSlider) {
+        super(document, key, label);
+        this.values = values;
+        this.index = defaultIndex;
+        this.defaultIndex = defaultIndex;
+        this.isSlider = isSlider;
+    }
+
+    public void cycle() {
+        this.index++;
+        if(index >= values.length) index = 0;
+        this.valueText = null;
+    }
+
+    @Override
+    public T getValue() {
+        return values[index];
+    }
+
+    @Override
+    public void setValue(T value) {
+        List<T> vList = Lists.newArrayList(values);
```

Comment:
By backing the implementation with a set or list I believe you can turn this right into a list.contains() -> do thing

Reply:
I don't think that kind of operation exists in lists/sets for `contains`, but I have decided to have the implementation rely on a list from the beginning rather than an array.